### PR TITLE
Chunk the cleanup queries to make sure they don't time out

### LINKED ISSUE #22243: files:cleanup run in deadlock for bigger installations
* maybe we should try to do a join and limit the statement to 1000 cleanups and do it simply more often

https://github.com/owncloud/core/blob/a6ade67dfbbae562634e9cce47d53f1aa78ad7e4/apps/files/command/deleteorphanedfiles.php#L55-L57

Just an idea how to make this command more useful and still clean things up.

### PR DESCRIPTION
**Pull request by nickvergessen**
- Chunked the select solution of [`lib/private/repair/repairinvalidshares.php`](https://github.com/owncloud/core/pull/22269/files?w=1#diff-edd1c1ac15a4d04eed6ab8a9af4e56a8)
- Adapted it to the other two instances where we have the same issue as well

Fix #22243 

@PVince81 @MorrisJobke

### CODE REVIEW

File: apps/files/command/deleteorphanedfiles.php [hunk 1]
```diff
@@ -50,13 +52,31 @@ protected function configure() {
...
+
+		$deleteQuery = $this->connection->getQueryBuilder();
+		$deleteQuery->delete('filecache')
+			->where($deleteQuery->expr()->eq('fileid', $deleteQuery->createParameter('objectid')));
 
-		$sql =
-			'DELETE FROM `*PREFIX*filecache` ' .
-			'WHERE NOT EXISTS ' .
-			'(SELECT 1 FROM `*PREFIX*storages` WHERE `storage` = `numeric_id`)';
+		$deletedInLastChunk = self::JUNK_SIZE;
```
Comment (PVince81) [inline apps/files/command/deleteorphanedfiles.php]: CHUNK which means a piece, not JUNK which means garbage :wink:

Comment (PVince81) [inline apps/files/command/deleteorphanedfiles.php]: Also see number 3 in http://www.urbandictionary.com/define.php?term=junk
You might want to refrain using that word here

Comment (nickvergessen) [inline apps/files/command/deleteorphanedfiles.php]: Woops, the commit message was right

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by karlitschek** [comment]
:+1:

**Pull request merged**